src/components/kalman_component/kalman.sh

```diff
@@ -21,9 +21,13 @@ setup_kf() {
     cp ${InversionPath}/src/notebooks/kf_notebook.ipynb ${RunDirs}/kf_inversions/
     sed -i 's|\/home\/ubuntu\/integrated_methane_inversion\/config.yml|'$ConfigPath'|g' ${RunDirs}/kf_inversions/kf_notebook.ipynb
 
-
     # Define Kalman filter update periods
-    python ${InversionPath}/src/components/kalman_component/make_periods_csv.py $StartDate $EndDate $UpdateFreqDays $RunDirs; wait
+    if "$MakePeriodsCSV"; then
+        python ${InversionPath}/src/components/kalman_component/make_periods_csv.py $StartDate $EndDate $UpdateFreqDays $RunDirs; wait
+    else
+        printf "Copying custom periods.csv to the run directory.\n"
+        cp $CustomPeriodsCSV ${RunDirs}/
+    fi
```

Comment:
Nice! I havent thought to much about this, but are there any assumptions built in to the KF assuming a uniform number of days?

Specifically, In the imi preview I think it assumes a certain number of days for calculating superobservation errors

Reply:
Yeah you're right! I made some changes in `imi_preview.py` so that the first period in periods.csv is used for the # of days to calculate the number of observations per period and A, but I'm not sure if that's the best way... https://github.com/geoschem/integrated_methane_inversion/pull/250/commits/dfeef5bde96bcbaf5d2109fd49e1420ce788a5e7